ModelViewIntentSample/search/src/main/java/com/novoda/movies/mvi/search/SearchActivity.kt

```diff
@@ -0,0 +1,42 @@
+package com.novoda.movies.mvi.search
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import com.novoda.movies.mvi.search.view.SearchInputView
+import com.novoda.movies.mvi.search.view.SearchResultsView
+import kotlinx.android.synthetic.main.activity_search.*
+
+internal class SearchActivity : AppCompatActivity() {
+
+    private lateinit var searchInput: SearchInputView
+    private lateinit var resultsView: SearchResultsView
+
+    lateinit var presenter: SearchResultsPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Injector().inject(this)
+        setContentView(R.layout.activity_search)
+        searchInput = search_input
+        resultsView = search_results
+    }
+
+    override fun onStart() {
+        super.onStart()
+        presenter.bind(searchInput, resultsView)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        presenter.unbind()
+    }
+
+    class Injector {
```

Comment:
@zegnus 
This is the interesting bit regarding the dependency injection:
Dependencies with `ApplicationScope` come from `searchActivity.application as Dependencies` where dependencies with `ActivityScope` come from the `SearchDependencyProvider`, which is created/destroyed together with the Activity. In case these would need to survive a configuration change, we could use an Android ViewModel to store them.

What do we achieve?
- By looking at the `SearchDependencyProvider` and it's dependencies it's very easy to understand where a dependency is coming from and what scope it has.
- Different scopes like `Singleton` aka. `ApplicationScope` and `ActivityScope` are easy to implement.
- If we want to replace a `xProvider` for instrumentation tests, we can declare an interface and overwrite the implementation with a test double. That obviously needs some more abstractions as we currently have here.